pennylane/measurements/__init__.py

```diff
@@ -11,8 +11,151 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""
-This module contains measurements supported by PennyLane.
+r"""
+This module contains all the measurements supported by PennyLane.
+
+Description
+-----------
+
+Measurements
+~~~~~~~~~~~~
+The :class:`MeasurementProcess` class serves as a base class for measurements, and is inherited
+from the :class:`SampleMeasurement`, :class:`StateMeasurement` and :class:`MeasurementTransform`
+classes. These classes are subclassed to implement measurements in PennyLane.
+
+* Each :class:`SampleMeasurement` subclass represents a sample-based measurement, which contains a
+  :meth:`SampleMeasurement.process_samples` method that processes the sequence of samples generated
+  by the device. This method should always have the same arguments:
+
+  * samples (Sequence[complex]): computational basis samples generated for all wires
+  * wire_order (Wires): wires determining the subspace that ``samples`` acts on
+  * shot_range (tuple[int]): 2-tuple of integers specifying the range of samples
+        to use. If not specified, all samples are used.
+  * bin_size (int): Divides the shot range into bins of size ``bin_size``, and
+        returns the measurement statistic separately over each bin. If not
+        provided, the entire shot range is treated as a single bin.
+
+  `See :class:`CountsMP` for an example.
```

Comment:
```suggestion
  See :class:`CountsMP` for an example.
```
Link doesnt work, maybe its just the `